app.js

```diff
@@ -0,0 +1,468 @@
+    // for the logical file IRI in multiple graphs, they will all be
+    // related to the same physical file IRI, so the array will always
+    // only contain 1 physical file IRI.
+    return result.results.bindings[0]['physicalFile'].value;
+  return null;
+}
+
+/**
+ * Converts a physical file IRI to a file path
+ *
+ * The URI of the stored file uses the share:// protocol and
+ * reflects the location where the file resides as a relative
+ * path to the share folder. E.g. share://uploads/my-file.pdf
+ * means the file is stored at /share/uploads/my-file.pdf.
+ * -- https://github.com/mu-semtech/file-service/blob/v3.3.0/README.md#description
+ */
+function filePathFromIRI(physicalFileIRI) {
+  return physicalFileIRI.replace(/^share:\/\//, '/share/');
+}
+
+/**
+ * Stores the result of a malware-scan in the database.
+ *
+ * A stix:MalwareAnalysis resource is stored in all graphs containing the
+ * supplied file IRI.
+ *
+ * Notes:
+ * - This function does not lookup and flag related file IRIs. If both the
+ *   logical/virtual file IRI and physical/stored file IRI need to be flagged,
+ *   call this function again for each IRI.
+ * - If fileIRI does not exist in any graph in the database, the returned
+ *   resource object will not have been inserted anywhere in the database.
+ *
+ * @param {String} fileIRI - IRI of the file to be flagged.
+ * @param {Object} stixMalwareAnalysis - The malware analysis details.
+ *        Properties: .analysisStarted: Timestamp of start of analysis.
+ *                    .analysisEnded  : Timestamp of end of analysis.
+ *                    .result : Usually one of the values from
+ *                        STIX 2.1 Malware Result Vocabulary malware-result-ov:
+ *                        "malicious", "suspicious", "benign" or "unknown".
+ *                        https://docs.oasis-open.org/cti/stix/v2.1/cs01/stix-v2.1-cs01.html#_dtrq0daddkwa
+ *                    .resultName : JSON string of array of viruses found.
+ * @return {Object} Properties:
+ *    .resourceObject: JavaScript object representation of the malware
+ *                     analysis resource object.
+ *    .databaseResponse: null if not inserted in any graph. Otherwise
+ *        .results.bindings[0].['callret-0'].value {String} Textual database
+ *            response mentioning the graphs in which the malware analysis
+ *            resource object was inserted (may still be 0, check response).
+ */
+async function storeMalwareAnalysis(fileIRI, stixMalwareAnalysis) {
+  const ret = {
+    resourceObject: undefined,
+    databaseResponse: undefined,
+  };
+
+  const malwareAnalysisId = uuid();
+  // TODO: Not http://data.gift/id/virus-scanner/analysis/1 ?
+  //       or: http://data.gift/services/id/virus-scanner/analysis/1 ?
+  const malwareAnalysisIri =
+    'http://data.gift/virus-scanner/analysis/id/'.concat(malwareAnalysisId);
+
+  let databaseResponse;
+  try {
+    databaseResponse = await update(`
```

Comment:
Side note:  extra reason why i am reluctant to expose as API: it uses a sudo query. so everyone with access could write to the db.